src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    public class Node<T> {
+        private T value;
+        private Node<T> previous;
+        private Node<T> next;
+
+        public Node(Node<T> previous, Node<T> next, T value) {
+            this.previous = previous;
+            this.next = next;
+            this.value = value;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, null, value);
+        if (isEmpty()) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.previous = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        ioobException(index);
+        Node<T> currentNode;
+        Node<T> nextNode = findByIndex(index);
+        if (index == 0) {
+            currentNode = new Node<>(null, nextNode, value);
+            nextNode.previous = currentNode;
+            head = currentNode;
+            size++;
+        } else {
+            currentNode = new Node<>(nextNode.previous, nextNode, value);
+            nextNode.previous.next = currentNode;
+            nextNode.previous = currentNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
```

Comment:
You can change local variable name. They must be more informative. Replace 't' to "value".
```suggestion
        for (T value : list) {
```